sycl/include/sycl/ext/intel/math/imf_integer_utils.hpp

```diff
@@ -0,0 +1,144 @@
+//==--------------------------- imf_integer_utils.hpp ----------------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+// C++ APIs for simple integer util funcitons
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+extern "C" {
+unsigned __imf_brev(unsigned);
+unsigned long long __imf_brevll(unsigned long long);
+unsigned __imf_byte_perm(unsigned, unsigned, unsigned);
+long long __imf_llmax(long long x, long long y);
+long long __imf_llmin(long long x, long long y);
+int __imf_max(int x, int y);
+int __imf_min(int x, int y);
+unsigned long long __imf_ullmax(unsigned long long x, unsigned long long y);
+unsigned long long __imf_ullmin(unsigned long long x, unsigned long long y);
+unsigned __imf_umax(unsigned x, unsigned y);
+unsigned __imf_umin(unsigned x, unsigned y);
+int __imf_clz(int);
+int __imf_clzll(long long);
+int __imf_ffs(int);
+int __imf_ffsll(long long);
+int __imf_mul24(int, int);
+int __imf_mulhi(int, int);
+long long __imf_mul64hi(long long, long long);
+int __imf_popc(unsigned);
+int __imf_popcll(unsigned long long);
+int __imf_rhadd(int, int);
+int __imf_hadd(int, int);
+unsigned __imf_sad(int, int, unsigned);
+unsigned __imf_uhadd(unsigned, unsigned);
+unsigned __imf_umul24(unsigned, unsigned);
+unsigned __imf_umulhi(unsigned, unsigned);
+unsigned long long __imf_umul64hi(unsigned long long, unsigned long long);
+unsigned __imf_urhadd(unsigned, unsigned);
+unsigned __imf_usad(unsigned, unsigned, unsigned);
+}
+
+namespace sycl {
+inline namespace _V1 {
+namespace ext::intel::math {
+
+template <typename Tp = unsigned> Tp brev(Tp x) { return __imf_brev(x); }
```

Comment:
From a design-perspective, I wonder if we should be doing something like
```suggestion
template <typename Tp = unsigned> std::enable_if_t<std::is_convertible_v<Tp, unsigned>, Tp> brev(Tp x) { return __imf_brev(x); }
```
Though the compiler will tell the user that `__imf_brev(x)` isn't valid if that condition doesn't currently hold, it is technically an internal detail and we could be warning user a level up.

Note that if we were to allow this, we would want separate template types for each argument on functions with multiple arguments, e.g. `byte_perm`, so the user could pass different (convertible) types to the functions. In some ways that already applies as currently the user should be seeing an error if they do something like `byte_perm(1u, (unsigned short)2u, 3u)`, while it should work just fine with `__imf_byte_perm(1u, (unsigned short)2u, 3u)` due to implicit conversion of non-template parameters.